packages/sync/src/offline/OfflineRestoreHandler.ts

```diff
@@ -48,14 +52,21 @@ export class OfflineRestoreHandler {
     this.offlineData.forEach(item => {
       const extensions = item.operation.extensions;
       const optimisticResponse = item.optimisticResponse;
-      this.apolloClient.mutate({
+      const mutationName = getMutationName(item.operation.query);
+      let updateFunction;
+      if (this.mutationCacheUpdates && mutationName) {
+        updateFunction = this.mutationCacheUpdates[mutationName];
+      }
+      const mutationOptions = {
         variables: item.operation.variables,
         mutation: item.operation.query,
         // Restore optimistic response from operation in order to see it
         optimisticResponse,
```

Comment:
this is just question from me. Given that we have the `optimisticResponse` here, can we actually use it to update the UI?

Reply:
Good question. The optimistic response is saved automatically and does not need any update method. Cache layer will also apply some heuristics to figure out affected queries (this will happen objects that are currently observed for current queries).
This is just an plain `Type` object response. We do not know what query this mutation will affect if there is a need for subscription deduplication etc.

UI is being updated thru cache subscriptions in showcase app - we need to make sure that this lands in the cache as it will be the only way to support many different query patterns on framework level + provide the ability to do edits on that objects as well. Skipping cache will cause many side effects and will require us to provide some custom handling. 

Technically we can use object enqueue listener instead of the update function, but we still need to map actual response from server anyway so it will mean doing twice the job. It is really good idea to consider:
Provide our own form of update functions that skip cache and simply deliver response to UI using some new customized observable stream. On startup we can rebuild entire stream and deliver those updates again however by this we going to need to largely reimplement optimistic response merging functionality from InMemory cache. That is not something I could get in short timeframe.